core/presentation/designsystem/src/main/java/com/nolawiworkineh/designsystem/components/TaskyToolBar.kt

```diff
@@ -0,0 +1,231 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.nolawiworkineh.designsystem.components.util.DropDownMenuItem
+
+
+@Composable
+fun TaskyTopAppBar(
+    title: String = "", // Default string title
+    showBackButton: Boolean = false,
+    isBlackToolBar: Boolean = true, // Determines whether toolbar is black
+    showEndIcon: Boolean = false,
+    isWhiteText: Boolean = false,
+    customTitle: (@Composable () -> Unit)? = null, // Optional custom title content
+    navigationIcon: (@Composable () -> Unit)? = null, // Leading icon
+    actionIcons: (@Composable RowScope.() -> Unit)? = null, // Action icons on the right
+    scrollBehavior: TopAppBarScrollBehavior = TopAppBarDefaults.enterAlwaysScrollBehavior(),
+    endIcon: (@Composable () -> Unit)? = null, // Dynamic end icon composable
+    menuItems: List<DropDownMenuItem> = emptyList(),
+    onMenuItemClick: (Int) -> Unit = {},
+    onBackClick: () -> Unit = {},
+) {
+    var isDropDownOpen by rememberSaveable { mutableStateOf(false) }
+
+    TopAppBar(
+
+        title = {
+            Box(
+                modifier = Modifier.fillMaxSize(),
+                contentAlignment = Alignment.Center
+            ) {
+                if (customTitle != null) {
+                    customTitle()
+                } else {
+                    Text(
+                        text = title,
+                        color = if (isWhiteText) Color.White else Color.Black,
+                        style = MaterialTheme.typography.titleLarge
+                    )
+                }
+            }
+        },
+
+        modifier = Modifier.fillMaxWidth(),
+
+        scrollBehavior = scrollBehavior,
+
+        colors = TopAppBarDefaults.topAppBarColors(
+            containerColor = if (isBlackToolBar) Color.Black else Color.White,
+        ),
+
+        navigationIcon = {
+            // If showBackButton is true, show the back button and handle onBackClick
+            if (showBackButton) {
+                IconButton(onClick = onBackClick) {
+                    Icon(
+                        imageVector = Icons.AutoMirrored.Filled.ArrowBack,
+                        contentDescription = "Back",
+                        tint = if (isWhiteText) Color.White else Color.Black
+                    )
+                }
+            } else if (navigationIcon != null) {
+                // If a custom navigation icon is provided, invoke it
+                navigationIcon()
+            }
+        },
+
+        actions = {
+            // If actionIcons are provided, invoke them (e.g., Save, Edit)
+            actionIcons?.invoke(this)
+
+            // Handle the dynamic end icon and menu items
+            if (showEndIcon) {
+                if (menuItems.isNotEmpty()) {  // Show dropdown only if there are menu items
```

Comment:
Could be chained in one if to remove one indentation of this already heavily intended code